sdk/typescript/src/providers/json-rpc-provider.ts

```diff
@@ -507,6 +526,9 @@ export class JsonRpcProvider extends Provider {
     digest: TransactionDigest
   ): Promise<SuiTransactionResponse> {
     try {
+      if (!isValidTransactionDigest(digest)) {
```

Comment:
We also need to do the same validation for `getTransactionWithEffectsBatch` on line 546